### PR TITLE
Improve support for patterns

### PR DESCRIPTION
**Pull request by tsegall**
- add schema & table pattern support to getTables()
 - add column pattern support to getColumns()
 - fix Catalog & Schema columns on getColumns()
 - add support for getSchema(catalog, schemaPattern)
 - flip error output to stderr

### CODE REVIEW

File: src/main/java/com/dbschema/mongo/MongoDatabaseMetaData.java [hunk 1]
```diff
@@ -61,18 +77,18 @@ public ResultSet getTables(String catalogName, String schemaPattern, String tabl
...
-      for (String cat : con.getService().getDatabaseNames()) {
-        for (String tableName : con.getService().getCollectionNames(cat)) {
-          resultSet.addRow(createTableRow(null, tableName));
+    Pattern pSchema = getPattern(schemaPattern);
+    Pattern pTable = getPattern(tableNamePattern);
+
+    for (String schema : con.getService().getDatabaseNames()) {
+      if (schemaPattern == null || schemaPattern.equals(schema) || (pSchema != null && pSchema.matcher(schema).matches())) {
+        for (String tableName : con.getService().getCollectionNames(schema)) {
+          if (tableNamePattern == null || tableNamePattern.equals(tableName) || (pTable != null && pTable.matcher(tableName).matches()))
```
Comment (kornilova203) [inline src/main/java/com/dbschema/mongo/MongoDatabaseMetaData.java:86]: I guess if we compile pattern then we should not check `tableNamePattern.equals(tableName)`

Comment (tsegall) [reply src/main/java/com/dbschema/mongo/MongoDatabaseMetaData.java]: So as suggested added support for getSearchStringEscape.  I also pulled the getPattern() method into a separate class as it should really be used in a few more spots - for example getPrimaryKeys (with tableNamePattern).  I also added a separate class for the associated junit tests.

**Comment by tsegall** [comment]
Any thoughts on the updated PR?

**Comment by kornilova203** [comment]
@tsegall, I'll review it tomorrow. Thank you!

**Comment by kornilova203** [comment]
Hello @tsegall,
Thank you for writing tests!
I don't see a reason why can't we support all cases e.g. `\_`. Also regex characters need to be escaped e.g. `hello%world.` should produce pattern `hello.*world\.` not this one: `hello.*world.`.

I fixed this problems locally. I'll merge your commit and push my fix. No actions required from your side.

**Pull request merged**